unity/Assets/Maroon/reusableScripts/Config/ConfigLoader.cs

```diff
@@ -0,0 +1,132 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using Newtonsoft.Json;
+using UnityEngine.Events;
+using SimpleJSON;
+using System.IO;
+using System.Linq;
+
+namespace Maroon.Config
+{
+    public class ConfigLoader : MonoBehaviour
+    {
+        public UnityEvent OnConfigLoaded = new UnityEvent();
+
+        protected string _currentConfigString;
+        protected int _currentConfigIndex;
+        protected Dictionary<string, string> _configs = new Dictionary<string, string>();
+
+        private static ConfigLoader _instance;
+
+        public string CurrentConfigString => _currentConfigString;
+
+        public int CurrentConfigIndex => _currentConfigIndex;
+        
+        public static ConfigLoader Instance => _instance;
+
+        private void Awake()
+        {
+            if (_instance == null)
+            {
+                _instance = this;
+            }
+            else if (_instance != this)
+            {
+                DestroyImmediate(this.gameObject);
+                return;
+            }
+
+            DontDestroyOnLoad(this.gameObject);
+        }
+
+        void Start()
+        {
+#if UNITY_WEBGL && !UNITY_EDITOR
+            StartCoroutine(LoadAllConfigs());
+#else
+            string basePath = Application.streamingAssetsPath + "/Config/3DMotionSimulation";
+            string[] txtFiles = Directory.GetFiles(basePath, "*.json");
+
+            _configs = new Dictionary<string, string>();
+            foreach (string file in txtFiles)
+            {
+                string json = File.ReadAllText(file);
+                string cutFilename = file.Replace(basePath + "\\", "");
+                cutFilename = cutFilename.Replace(".json", "");
+                
+                _configs.Add(cutFilename, json);
+            }
+
+            ChangeConfig("Default");
+#endif
+        } 
+
+
+#if UNITY_WEBGL && !UNITY_EDITOR
+        IEnumerator LoadAllConfigs()
+        {
+            string basePath = "http://localhost:8000/StreamingAssets/Config/3DMotionSimulation/";
```

Comment:
only works for port 8000 now, that somehow seems like it should be changed; also maybe someone wants to use HTTPS, maybe Maroon is in a subdirectory and not in the root directory of the website.